realm/realm-annotations-processor/src/test/resources/io/realm/some_test_BooleansRealmProxy.java

```diff
@@ -502,9 +502,15 @@ public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass()) return false;
         some_test_BooleansRealmProxy aBooleans = (some_test_BooleansRealmProxy)o;
 
-        String path = proxyState.getRealm$realm().getPath();
-        String otherPath = aBooleans.proxyState.getRealm$realm().getPath();
+        BaseRealm realm = proxyState.getRealm$realm();
+        BaseRealm otherRealm = aBooleans.proxyState.getRealm$realm();
+        String path = realm.getPath();
+        String otherPath = otherRealm.getPath();
         if (path != null ? !path.equals(otherPath) : otherPath != null) return false;
+        if (realm.isFrozen() != otherRealm.isFrozen()) return false;
+        if (!realm.sharedRealm.getVersionID().equals(otherRealm.sharedRealm.getVersionID())) {
```

Comment:
IMO Yes, but it is still an open question exactly what equality means in the context of a Realm object.

I.e. Is `Object 1:V1` equal to `Object: V2`?

- if the version is different, but all properties are the same (note, checking this might be extremely expensive if there are links involved)?
- If the version is different and properties have changed?
- What if objects are frozen?

Since this is the auto-generated equals method I opted for the conservative approach, i.e. the objects are only equal if they have the same "state", which involves checking both version and frozen state, even if they have the same property values they are still considered "not-equal" if the version is different.

If people want to loosen this, they can implement equals themselves, which will override this implementation.

But this was also one of the open questions I raised in the original post about this PR, so any input to these semantics this would be great.

Reply:
I don't think there's a right decision here since it's subject to the user intention, we should stick with a coherent default implementation and let the user customize it as necessary.

From my perspective, two managed objects are equals if they reference the same `ObjKey` and have the same version ID, this is quick to check (even with links). Implementing this as the default behaviour will allow the user to override it with a version of `equal` based on the content if he wants.

Reply:
So you would say that a frozen object is equal to the live version if they have the same version?

I guess I would just worry that one object accidentally being equal to another could cause problems, e.g. if used as a key in a HashMap or similar:

```
// Example
Person person = realm.where(Person.class).findFirst();
Person frozenPerson = person.freeze();
person.equals(frosenPerson); // true
realm.refresh();
person.equals(frosenPerson); // potentially false
```

Reply:
but that's the usage/choice of a mutable key for a Map which is a bad practice (since the hash code changes ...)

Reply:
With Core 6 I would think we would change `RealmObject.hashCode()` to use the `ObjKey` since hash codes are allowed to be the same even if `equals` returns false